source/generators/SingleParticle.cc

```diff
@@ -110,80 +107,47 @@ void SingleParticle::SetParticleDefinition(G4String particle_name)
 
 void SingleParticle::GeneratePrimaryVertex(G4Event* event)
 {
-  // Generate an initial position for the particle using the geometry
-  G4ThreeVector position = geom_->GenerateVertex(region_);
-
-  // Particle generated at start-of-event
-  G4double time = 0.;
-
-  // Create a new vertex
-  G4PrimaryVertex* vertex = new G4PrimaryVertex(position, time);
-
   // Generate uniform random energy in [E_min, E_max]
-  G4double kinetic_energy = RandomEnergy();
-
-  // Generate random direction by default
-  G4ThreeVector momentum_direction_ = G4RandomDirection();
+  G4double kinetic_energy = nexus::RandomEnergy(energy_max_, energy_min_);
```

Comment:
There is already a `using namespace nexus;` near the top of this file, so the the `nexus::` isn't needed here, though you could leave it in, if you think that it adds some useful clarity.